### PR TITLE
Reformat to LLVM style add style config files...

### PR DESCRIPTION
**Pull request by rocky**
.editorconfig and .clang-format

### CODE REVIEW

**Comment by rocky** [comment]
Reviewers and @philippemilink you have comments or questions on this, let me know. If I don't hear back in a week or so, I'll just merge this.

File: lib/paranoia/paranoia.c [hunk 1]
```diff
@@ -325,49 +317,42 @@ i_paranoia_overlap2(int16_t *buffA,int16_t *buffB,
...
+  else if ((flagB[posB] & FLAGS_UNREAD) == 0)
+    ret = i_paranoia_overlap2(cv(A), iv(B), flagA, flagB, posA, posB, cs(A),
+                              is(B), begin, end);
 
   /* Small matching runs could just be coincidental.  We only consider this
    * a real match if it's long enough.
    */
   if (ret > MIN_WORDS_SEARCH) {
-    *offset=+(posA+cb(A))-(posB+ib(B));
+    *offset = +(posA + cb(A)) - (posB + ib(B));
```
Comment (philippemilink) [inline lib/paranoia/paranoia.c:342]: I'm surprised by the `+` between `=` and `(`: either it is normal and then the `+` is useless (I guess?) or it should be `+=`

**Review by philippemilink — COMMENTED** [review]
Debian package builds fine with this patch applied. 

I'm not a fan of spaces between `#` and `define`, but I guess that's just a matter of habits.

Expect the comment about the `= +`, LGTM after a quick look (but I'm not familiar at all with the code of this project!).

**Comment by rocky** [comment]
> Debian package builds fine with this patch applied.
> 
> I'm not a fan of spaces between `#` and `define`, but I guess that's just a matter of habits.
> 

Noted that this is not universally liked. When clang-format runs it does remove this. If others mentions or this is auto-formatted in the future it will disappear

File: lib/paranoia/paranoia.c (hunk 1 above)
Comment (rocky) [reply lib/paranoia/paranoia.c:342]: I've traced this change back to the SVN change from r36 to r37 which has this generic commit

```
------------------------------------------------------------------------

r37 | xiphmont | 1999-08-15 18:29:59 -0400 (Sun, 15 Aug 1999) | 6 lines



Imported cdparanoia III alpha 9.3 into CVS
```

The 9.2 line  was: 

```python
    *offset=(posA+isort_begin(A))-(posB+isort_begin(B));
```

Given the lack of information about the motivation for the "+" I'll remove it so it does not cause others to wonder about it.

**Pull request merged**